core/client/fs/src/main/java/alluxio/client/file/FileSystem.java

```diff
@@ -145,10 +147,15 @@ public static FileSystem create(FileSystemContext context) {
       Object[] ctorArgs = new Object[] {context};
       FileSystem fs =
           (FileSystem) CommonUtils.createNewClassInstance(fsClass, ctorArgClasses, ctorArgs);
-      if (context.getClusterConf().getBoolean(PropertyKey.USER_LOCAL_CACHE_ENABLED)) {
-        return new LocalCacheFileSystem(fs);
+
+      boolean dataCache = context.getClusterConf().getBoolean(PropertyKey.USER_LOCAL_CACHE_ENABLED);
+      boolean metadataCache =
+          context.getClusterConf().getBoolean(PropertyKey.USER_METADATA_CACHE_ENABLED);
+      if (dataCache) {
+        return metadataCache ? new MetadataCachingLocalCacheFileSystem(fs, context)
+            : new LocalCacheFileSystem(fs);
```

Comment:
Instead of having a separate `MetadataCachingLocalCacheFileSystem`, can we compose them like `LocalCacheFileSystem(MetadataCachingBaseFileSystem)`?